### PR TITLE
SDK-1701: Created NO_INSTALL_PACKAGES setting for composer

### PR DESCRIPTION
**Pull request by vol4onok**
https://spryker.atlassian.net/browse/SDK-1701
- Created configurable env setting for composer install.

### CODE REVIEW

**Comment by dereuromark** [comment]
Nice, this is probably speeding it up quite a bit, right?

**Comment by vol4onok** [comment]
> 

yes, it should save some time.

File: src/Upgrade/Infrastructure/Configuration/ConfigurationProvider.php [hunk 1]
```diff
@@ -58,6 +58,11 @@ class ConfigurationProvider implements ConfigurationProviderInterface
      */
     protected const DEFAULT_IS_PR_AUTO_MERGE_ENABLED = false;
 
+    /**
+     * @var bool
+     */
+    protected const NO_INSTALL_PACKAGES = false;
```
Comment (pavelmaksimov25) [inline src/Upgrade/Infrastructure/Configuration/ConfigurationProvider.php:64]: `COMPOSER_INSTALL_DEPENDENCIES`
Afaik, composer should use `--no-install` option by default.

Comment (vol4onok) [reply src/Upgrade/Infrastructure/Configuration/ConfigurationProvider.php:64]: I would keep existing behavior by default. but ok

**Review by pavelmaksimov25 — APPROVED** [review]

File: src/Upgrade/Infrastructure/Configuration/ConfigurationProvider.php [hunk 2]
```diff
@@ -68,6 +73,16 @@ public function getUpgradeStrategy(): string
...
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @return bool
+     */
+    public function getComposerInstallDependencies(): bool
+    {
+        return (bool)getenv('COMPOSER_INSTALL_DEPENDENCIES') ?: static::COMPOSER_INSTALL_DEPENDENCIES;
```
Comment (VladislavStrelchenko) [inline src/Upgrade/Infrastructure/Configuration/ConfigurationProvider.php:83]: Please add information about the configuration `COMPOSER_INSTALL_DEPENDENCIES` to README.md

Comment (vol4onok) [reply src/Upgrade/Infrastructure/Configuration/ConfigurationProvider.php:83]: +

File: README.md [hunk 3]
```diff
@@ -103,6 +103,14 @@ The Upgrader updates projects via one of the following approaches:
 ```bash
 export UPGRADE_STRATEGY=composer
 ```
+#### Composer update strategy
+Composer install|update uses two strategies.
+By default composer updates only lock file.
+
+* To enable update composer with packages:
+```bash
+export COMPOSER_INSTALL_DEPENDENCIES=false
```
Comment (VladislavStrelchenko) [inline README.md:112]: Is it correct behavior? 

`To enable update composer with packages = COMPOSER_INSTALL_DEPENDENCIES = false`

From my perspective better:

`To enable update composer with packages = COMPOSER_INSTALL_DEPENDENCIES = true`

Can we change the behavior to the second version?

**Comment by VladislavStrelchenko** [comment]
you need to cover new functionality in updated classes

File: README.md (hunk 3 above)
Comment (vol4onok) [reply README.md:112]: it's correct. see comment above

Comment (VladislavStrelchenko) [reply README.md:112]: Yes you are right 
By default: we have use `--no-install flag 
That's mean: by default, config `COMPOSER_INSTALL_DEPENDENCIES` = false
and for enable it we will add `export COMPOSER_INSTALL_DEPENDENCIES = true`

**Comment by DmytroKlymanSpryker** [comment]
```
FILE: ...lication/Strategy/Common/Step/ComposerLockComparatorStepTest.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 136 | ERROR | Doc Block param variable `$noInstall` should be
     |       | `$withDependencies`
     |       | (Spryker.Commenting.DocBlockParam.VariableWrong)
----------------------------------------------------------------------


FILE: ...e/PackageManager/CommandExecutor/ComposerCommandExecutorTest.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 83 | ERROR | Doc Block param variable `$noInstall` should be
    |       | `$withDependencies`
    |       | (Spryker.Commenting.DocBlockParam.VariableWrong)
----------------------------------------------------------------------

```

**Review by VladislavStrelchenko — APPROVED** [review]

**Pull request merged**